src/util/AbstractHelper.sol

```diff
@@ -0,0 +1,337 @@
+    }
+
+    /**
+    @notice Sells DBR on behalf of the caller and uses the proceeds along with DOLA from the caller to repay debt.
+    @dev The caller is unlikely to spend all of the DOLA they make available for the function call
+    @param market The market the user wishes to repay debt in
+    @param dolaAmount The maximum amount of dola debt the user is willing to repay
+    @param minDolaFromDbr The minimum amount of DOLA the caller expects to get in return for selling their DBR
+    @param dbrAmountToSell The amount of DBR the caller wishes to sell
+    */
+    function sellDbrAndRepayOnBehalf(IMarket market, uint dolaAmount, uint minDolaFromDbr, uint dbrAmountToSell) public {
+        uint dbrBal = DBR.balanceOf(msg.sender);
+
+        //If user has less DBR than ordered, sell what's available
+        if(dbrAmountToSell > dbrBal){
+            DBR.transferFrom(msg.sender, address(this), dbrBal);
+            _sellExactDbr(dbrBal, minDolaFromDbr);
+        } else {
+            DBR.transferFrom(msg.sender, address(this), dbrAmountToSell);
+            _sellExactDbr(dbrAmountToSell, minDolaFromDbr);
+        }
+
+        uint debt = market.debts(msg.sender);
+        uint dolaBal = DOLA.balanceOf(address(this));
+        
+        //If the debt is lower than the dolaAmount, repay debt else repay dolaAmount
+        uint repayAmount = debt < dolaAmount ? debt : dolaAmount;
+
+        //If dolaBal is less than repayAmount, transfer remaining DOLA from user, otherwise transfer excess dola to user
+        if(dolaBal < repayAmount){
+            require(repayAmount <= dolaAmount + dolaBal, "Repay amount exceeds dola amount");
+            DOLA.transferFrom(msg.sender, address(this), repayAmount - DOLA.balanceOf(address(this)));
+        } else {
+            DOLA.transfer(msg.sender, dolaBal - repayAmount);
+        }
+
+        //Repay repayAmount
+        DOLA.approve(address(market), repayAmount);
+        market.repay(msg.sender, repayAmount);
+    }
+
+    /**
+    @notice Sells DBR on behalf of the caller and uses the proceeds along with DOLA from the caller to repay debt, and then withdraws collateral
+    @dev The caller is unlikely to spend all of the DOLA they make available for the function call
+    @param market Market the user wishes to repay debt in
+    @param dolaAmount Maximum amount of dola debt the user is willing to repay
+    @param minDolaFromDbr Minimum amount of DOLA the caller expects to get in return for selling their DBR
+    @param dbrAmountToSell Amount of DBR the caller wishes to sell
+    @param collateralAmount Amount of collateral to withdraw
+    @param deadline Deadline of the signature
+    @param v V parameter of the signature
+    @param r R parameter of the signature
+    @param s S parameter of the signature
+    */
+    function sellDbrRepayAndWithdrawOnBehalf(
+        IMarket market, 
+        uint dolaAmount, 
+        uint minDolaFromDbr,
```

Comment:
minDolaFromDbr is sensitive. UI should provide a reasonable value to prevent sandwich attacks. Comments should make note of it for users who interact directly with the contracts.

Reply:
Fixed in [74e7e73](https://github.com/InverseFinance/FiRM/pull/33/commits/74e7e733169f454ecae4b9c771f033b0cdad8d84)